doc/ADDB_Monitoring.rst

```diff
@@ -0,0 +1,281 @@
+
+#. Create ADDB monitor, add it to the global list of monitors. 
+
+#. Define the type of addb record that it will generate. 
+
+#. Get the statistics information from these addb records periodically. 
+
+#. Send this statistical information to the endpoint where stats service is running as addb records & to the endpoint where addb service is running if the node is a client or to the addb stob if the node is server periodically.
+
+**Statistical monitoring of addb records that do not contain statistical information in them** 
+
+Following steps show how an addb monitor collects statistical information on a particular node(client/server) from addb records and send it to stats service as addb records:
+
+#. Create ADDB monitor, add it to the global list of monitors. 
+
+#. Define the type of addb record that it will generate. 
+
+#. Continuously compute statistics from the monitored addb records. 
+
+#. Send this statistical information to the endpoint where stats service is running as addb records & to the endpoint where addb service is running if the node is a client or to the addb stob if the node is server periodically.     
+
+**Exceptional conditions monitoring**
+
+Exceptional conditions such as failures, overflows, etc. could be generated inside monitoring(exceptions occurred as a result of interpreting the statistical information generated after monitoring addb records) or outside monitoring (other sub-system failures). Following steps are to be taken:
+
+#. Generate the exception description fop. 
+
+#. Post this fop to a local HA component.  
+
+**Building a cluster wide global & local state in memory on a node where stats service is running**  
+
+#. Create in-memory state structure of the cluster on this node. 
+
+#. Receive statistical summary addb records from all the node. 
+
+#. Update the state with the information in these latest addb records.
+
+**Query for some state information to the stats service**
+
+#. Construct & send a request fop for specific or complete state information to the stats service & wait for reply. 
+
+#. Stats service checks for requesting information, gathers it in reply fop & sends it back to the node from where request was initiated.
+
+ Failures
+==========
+
+Following failure cases are listed along with their handling mechanism: 
+
+- A failure to construct new state on the node where the stats service runs would return the previous state to the node that requested this state information during this duration.
+
+- Exceptional conditions are reported to local HA component using a fop, a failure of receiving a fop by local HA component can happen, this would mean that some exceptional conditions can go unnoticed by local HA component. This type of failure is ignored.
+
+ Rationale
+============
+
+The existing ADDB implementation and the newly developed tracing subsystem contributed greatly to the requirement to use C macro interfaces with compile time validation. 
+
+ Compatibility
```

Comment:
Please conform with the original organization of titles in different levels.

Reply:
Incorporated.